trove/util/trove_params.py

```diff
@@ -107,13 +108,20 @@ def _gather_attrpaths(cls, queryparams: _qp.QueryparamDict, shorthand: rdf.IriSh
                         f'expected "fields[TYPE]" (with exactly one non-empty bracketed segment)'
                         f' (got "{_param_name}")'
                     )
+                if _typenames == '*':
+                    wildcard_paths.extend(
+                        parse_propertypath(_path_value, shorthand)
+                        for _path_value in _qp.split_queryparam_value(_param_value)
+                    )
                 else:
                     for _type in _qp.split_queryparam_value(_typenames):
                         _type_iri = shorthand.expand_iri(_type)
                         _requested[_type_iri].extend(
                             parse_propertypath(_path_value, shorthand)
                             for _path_value in _qp.split_queryparam_value(_param_value)
                         )
+            if wildcard_paths:
+                _requested['*'].extend(wildcard_paths)
```

Comment:
hmm, should be able to add this logic without duplicating the `parse_propertypath`/`split_queryparam_value` comprehension (or adding a separate `wildcard_paths` list)
```py
                    for _type in _qp.split_queryparam_value(_typenames):
                        _type_key = (
                            GLOB_PATHSTEP
                            if _type == GLOB_PATHSTEP
                            else shorthand.expand_iri(_type)
                        )
                        _requested[_type_key].extend(...
```
(this is a bit more lenient; allows something like `fields[*,Foo]=...`, but that's probably ok)